operator/pkg/tasks/init/cert.go

```diff
@@ -101,6 +104,31 @@ func runCATask(kc *certs.CertConfig) func(d workflow.RunData) error {
 		if kc.CAName != "" {
 			return fmt.Errorf("this function should only be used for CAs, but cert %s has CA %s", kc.Name, kc.CAName)
 		}
+
+		customCertConfig := data.CustomCertificate()
+		if kc.Name == constants.CaCertAndKeyName && customCertConfig.APIServerCACert != nil {
+			secretRef := customCertConfig.APIServerCACert
+			klog.V(4).InfoS("[certs] Loading custom CA certificate", "secret", secretRef.Name, "namespace", secretRef.Namespace)
+
+			certData, keyData, err := loadCACertFromSecret(data.RemoteClient(), secretRef)
+			if err != nil {
+				return fmt.Errorf("failed to load custom CA certificate: %w", err)
+			}
+
+			klog.V(2).InfoS("[certs] Successfully loaded custom CA certificate", "secret", secretRef.Name)
+
+			customKarmadaCert := &certs.KarmadaCert{
+				PairName: kc.Name,
+				CAName:   kc.CAName,
+				Cert:     certData,
+				Key:      keyData,
+			}
+
+			data.AddCert(customKarmadaCert)
+			klog.V(2).InfoS("[certs] Successfully added custom CA certificate to cert store", "certName", kc.Name)
+			return nil
+		}
```

Comment:
How about extracting two functions: one to determine whether it is a custom CA, and another to load the `karmadaCert` using `secretRef`. This may enhance the extensibility of the code.

Reply:
I think this is fine. It's  a one liner that's very specific to the API Server CA. It's not general.

Reply:
> I think this is fine. It's a one liner that's very specific to the API Server CA. It's not general.

Get it~ 
Could we abstract the process of loading the `karmadaCert` by `secretRef` into a function? This way, if there's a need for custom other certificates in the future, this function could be used universally.

Reply:
Ok. Will make the change.

Reply:
Not sure what you mean.  That's actually already done in the `loadCACertFromSecret` function.

Reply:
I mean L111-L125 can be extracted into a function. However, this is not important either; we can consider it when there is a real need.

Reply:
Yeah. I agree with you on not abstracting the logic prematurely.